src/shared/image_builder_lib.sh

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env bash
+
+CICD_TOOLS_IMAGE_BUILDER_LOADED=${CICD_TOOLS_IMAGE_BUILDER_LOADED:-1}
+
+if [[ "$CICD_TOOLS_IMAGE_BUILDER_LOADED" -eq 0 ]]; then
+  return 0
+fi
+
+if [ -z "$CICD_TOOLS_SCRIPTS_DIR" ]; then
+  echo "scripts directory not defined, please load through main.sh script" >&2
+  return 1
+fi
+
+cicd_tools::debug "loading image builder library"
+
+readonly CICD_TOOLS_IMAGE_BUILDER_REDHAT_REGISTRY='registry.redhat.io'
+readonly CICD_TOOLS_IMAGE_BUILDER_QUAY_REGISTRY='quay.io'
+readonly CICD_TOOLS_IMAGE_BUILDER_QUAY_EXPIRE_TIME=${CICD_TOOLS_IMAGE_BUILDER_QUAY_EXPIRE_TIME:-3d}
+readonly CICD_TOOLS_IMAGE_BUILDER_QUAY_USER="${CICD_TOOLS_IMAGE_BUILDER_QUAY_USER:-$QUAY_USER}"
+readonly CICD_TOOLS_IMAGE_BUILDER_QUAY_PASSWORD="${CICD_TOOLS_IMAGE_BUILDER_QUAY_PASSWORD:-$QUAY_TOKEN}"
+readonly CICD_TOOLS_IMAGE_BUILDER_REDHAT_USER="${CICD_TOOLS_IMAGE_BUILDER_REDHAT_USER:-$RH_REGISTRY_USER}"
+readonly CICD_TOOLS_IMAGE_BUILDER_REDHAT_PASSWORD="${CICD_TOOLS_IMAGE_BUILDER_REDHAT_PASSWORD:-$RH_REGISTRY_TOKEN}"
+readonly CICD_TOOLS_IMAGE_BUILDER_DEFAULT_BUILD_CONTEXT='.'
+readonly CICD_TOOLS_IMAGE_BUILDER_DEFAULT_CONTAINERFILE_PATH='Dockerfile'
+
+cicd_tools::image_builder::build_and_push() {
+  cicd_tools::image_builder::build || return 1
+  if cicd_tools::image_builder::is_change_request_context; then
+  cicd_tools::image_builder::push || return 1
+  fi
+}
+
+cicd_tools::image_builder::build() {
+
+  local containerfile build_context image_name image_tags default_image_name
+  declare -a label_params image_tag_params build_arg_params
+
+  containerfile="$(cicd_tools::image_builder::get_containerfile)"
+  build_context="$(cicd_tools::image_builder::get_build_context)"
+  image_name="$(cicd_tools::image_builder::_get_image_name)" || return 1
+  default_image_name=$(cicd_tools::image_builder::get_full_image_name) || return 1
+
+  if ! [ -r "$containerfile" ]; then
+  cicd_tools::err "Containerfile '$containerfile' does not exist or is not readable!"
+  return 1
+  fi
+
+  image_tag_params=('-t' "$default_image_name")
+
+  if ! cicd_tools::image_builder::is_change_request_context; then
+    for additional_tag in $(cicd_tools::image_builder::get_additional_tags); do
+      image_tag_params+=('-t' "${image_name}:${additional_tag}")
+    done
+  fi
+  
+  for label in $(cicd_tools::image_builder::get_labels); do
+  label_params+=('--label' "${label}")
```

Comment:
There's a few lines in this file that need an indent fix... double check check everything below here :)